index.js

```diff
@@ -136,49 +192,73 @@ function nowServing(line) {
 }
 
 const line = []
+let ticketNum = 0
+
+// code
+function takeATicketNumber (line) {
+
+  line.push(line.length + 1)
+  ticketNum += 1
+
+  return `Welcome. You are ticket number ${ticketNum}`
+}
+
+
 
 // ***** Scope & Closures - Question 1 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log("%c----------", "color: red") 
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc () {
+  let tickNum = 0
+  return function takeATicketNumberFunc (newLine) {
+    newLine.push(newLine.length + 1)
+    // tickNum = newLine.length doesn't work, why?
+    tickNum += 1
+
+    return `Welcome. You are ticket number ${tickNum}`
+  }
+}
```

Comment:
`tickNum = newLine.length` doesn't work because using the `newLine.length` way, the tickets don't get updated past a certain number. The test pushes two people into the line and gives them ticket 1 and 2. Then we serve the two front people (so the people who have tickets 1 and 2) and remove them from the line. Tickets 1 and 2 should be used up now, the next available ticket should be 3, and our line is empty. However, when we call `takeATicketNumberFunc(newLine)` for the last time, you use `newLine.length + 1` to compute the next ticket number, which is 1 because our line is empty.